File: 프로그래머스/0/181934. 조건 문자열/조건 문자열.js
function solution(ineq, eq, n, m) {
    let answer = 0;
    
    if (ineq === "<") {
        // ineq가 <인 경우
        if (eq === "=") {
            // eq가 = 인 경우
            if (n - m > 0) {
                answer = 0;
            } else {
                answer = 1;
            }
        } else {
            // eq가 ! 인 경우
            if (n - m < 0) {
                answer = 1;
            } else {
                answer = 0;
            }
        }
    } else {
        // ineq가 >인 경우
        if (eq === "=") {
            // eq가 = 인 경우
            if (n - m >= 0) {
                answer = 1;
            } else {
                answer = 0;
            }
        } else {
            // eq가 ! 인 경우
            if (n - m > 0) {
                answer = 1;
            } else if (n - m === 0) {
                answer = 0;
            } else {
                answer = 0;
            }
        }
    }
    return answer;
}

File: 프로그래머스/unrated/181934. 조건 문자열/조건 문자열.java
class Solution {
    public int solution(String ineq, String eq, int n, int m) {
        int answer = 0;
        
        if(ineq.equals("<")) {
        // ineq가 <인 경우
			 if(eq.equals("=")) {
             // eq가 = 인경우
				 if(n - m > 0) {
					 answer = 0;
				 } 
				 else {
					 answer = 1;
				 }
			 }
			 else {
             // eq가 ! 인경우
				 if(n - m < 0) {
					 answer = 1;
				 }
				 else {
					 answer = 0;
				 }
			 }
		}
		else {
        // ineq가 >인 경우
			if(eq.equals("=")) {
            // eq가 = 인경우
				if(n - m >= 0) {
					answer = 1;
				}
				else {
					answer = 0;
				}
			}
			else {
            // eq가 ! 인경우
				if(n - m > 0) {
					answer = 1;
				}
                else if(n - m == 0) {
                    answer = 0;
                }
				else {
					answer = 0;
				}
			}
		}
        return answer;
    }
}